examples/minimal-mdns/advertiser.cpp

```diff
@@ -124,10 +123,15 @@ bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier,
         gOptions.productId = Optional<uint16_t>::Value(static_cast<uint16_t>(atoi(aValue)));
         return true;
     case kOptionCommisioningMode:
-        gOptions.commissioningMode = true;
-        return true;
-    case kOptionCommisioningModeOpenWindow:
-        gOptions.commissioningModeOpenWindow = true;
+        cm = static_cast<uint8_t>(atoi(aValue));
```

Comment:
nit: prefer strcmp with values. I would rather read `--commissioning-mode basic` or `--commissioning-mode enhanced` than `--commissioning-mode 1` or `--commissioning-mode 2`. A log saying 'invalid value' is better than return without change (I would not know that `--commissioning-mode 123` does not work)

I believe eventually the minimal-dns examples may go away though, so will merge even if this is not done in this PR.

Reply:
leaving this for now